### PR TITLE
Master datetime js adt

### PR DESCRIPTION
**Pull request by Elkasitu**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Elkasitu** [comment]
@ged-odoo Can I ask you or someone from your team to take a look at this?

**Comment by ged-odoo** [comment]
I think @Polymorphe57 is the most qualified for this!

File: addons/web/static/lib/py.js/lib/py_extras.js [hunk 1]
```diff
@@ -112,6 +112,25 @@ function ymd2ord(year, month, day) {
...
+function get_quarter(year, month) {
+    var quarter_number = get_quarter_number(month);
+    var month_from = ((quarter_number - 1) * 3) + 1
+    var date_from = {year: year, month: month_from, day: 1}
+    var date_to = {year: year, month: month_from + 2, day: days_in_month(year, month)}
+    return [date_from, date_to];
+}
+
+function get_day_of_week(year, month, day) {
+    // Since JavaScript is a piece of garbage, months start at 0
```
Comment (ged-odoo) [inline addons/web/static/lib/py.js/lib/py_extras.js]: Heretic! Everything should be 0-indexed, not 1-indexed!

Comment (Elkasitu) [reply addons/web/static/lib/py.js/lib/py_extras.js]: Why is it not the same for `day` then? :thinking:

**Pull request merged**